File: aire-api/src/main/java/io/aire/designer/Element.java
package io.aire.designer;

import java.util.Collection;
import java.util.List;

public interface Element extends PropertyAware, Alignable {

  enum Category {
    Edge,
    Vertex;
  }

  String getId();

  void setId(String id);

  ElementType getType();

  void setType(ElementType type);

  Image addImage(Image image);

  Image removeImage(Image image);

  boolean hasImageWithRole(ImageType imageType);

  List<Image> getImages();

  Category getCategory();

  ElementStyle getStyle();

  Collection<ElementOverlay> getOverlays();
}

File: aire-core/src/main/frontend/application/aire/designer/core/element.ts
import {Image, ImageType} from "@aire/designer/core/image";

export enum Category {
  Palette = "palette",
  Element = "element"
}


export interface ImagePayload {
  width: number;
  height:number;
  type: string;
  source: string;

}
export interface ElementPayload {
  category : string;
  images : ImagePayload[];
}

export interface Element {
  category: Category;
  images: Map<String, Image>;

  addImage(image:Image): void;

  getImage(key: ImageType) : Image;

}